### ISSUE TITLE
Please remove embedded libs

### ISSUE THREAD

**Issue by pcav** [Gaia3D/OnMapLoader#5]
Plugins rely on external libs instead of internal ones. Having multiple copies of the libs can lead to problems.
Thanks.

**Comment by jangbi882** [comment]
@pcav 
Thank you for response!

My Plugin needed PyPDF2 Library for extract image clips in PDF file.
But original QGIS does not have this library.

Please tell me how to solve this problem.

**Comment by pcav** [comment]
You have two ways:
* ask the user to install (many plugins require external dependencies)
^ help us including the lib in osgeo4w and therefore in the standalone installer

**Comment by jangbi882** [comment]
@pcav 
But the pip command in the QGIS Windows distribution does not work.

**Comment by pcav** [comment]
than it should be fixed; did you open a ticket? Can you help solving it?

**Comment by jangbi882** [comment]
I already open ticket.
https://issues.qgis.org/issues/15944

**Comment by jangbi882** [comment]
I usually fix this problem using command.
```
python -m pip install --upgrade pip
```

**Comment by jangbi882** [comment]
But this command not work at some version .
I solved this problem uninstall and reinstall pip module in my current version 2.18.11.

**Comment by pcav** [comment]
Please check in latest version.
Thanks.

Il 1 febbraio 2018 14:20:21 WET, BJ Jang <notifications@github.com> ha scritto:
>But this command some version not work.
>I solved this problem uninstall and reinstall pip module in my current
>version 2.18.11.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/Gaia3D/OnMapLoader/issues/5#issuecomment-362279397

-- 
Sorry for being short

**Comment by jangbi882** [comment]
Yes, pip is work well on version 2.18.15. Thanks.
But the policy of letting you install additional libraries by each user is hard to follow.
I will give up registration of plugin repository. It takes tooooo long and is a difficult process.

**Comment by jangbi882** [comment]
@pcav 
Windows QGIS 2.18.16's pip command still not work. The packaging has problem.
Above QGIS 2.18.15's pip command was solved by my colleagues.
I was mistaken.

**Comment by pcav** [comment]
Strange answer. Could you please help fixing the pip issue upstream?
Please remember we have to manage about 700 plugins, all on a voluntary basis.
You are one of the very few to complain.

**Comment by jangbi882** [comment]
I understand your policies and agree completely.
I am also always thankful for your hard work.

However, it took more than two months to get a proper response to this case, and I was tired.

I will make a new ticket for the pip problem when I have a little free time.
In addition, I will make an unofficial plugin repository and distribute my plugins.

Do not forget that many users, including myself, appreciate the hard work of QGIS developers and management.
Thank you so much and I love all QGIS Guys~

**Comment by pcav** [comment]
All the best.
